### ISSUE TITLE
CA1508 False positive for Activator.CreateInstance

### ISSUE THREAD

**Issue by meziantou** [dotnet/roslyn-analyzers#4548]
### Analyzer

**Diagnostic ID**: CA1508

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.101](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug

`Activator.CreateInstance` can return `null` when used with a `Nullable<T>` type.

````
var value = Activator.CreateInstance(typeof(int?));
if (value is null) // CA1508 'value is null' is always 'false'
{
    Console.WriteLine("null");
}
````

In the previous code `value is null` is true. Whereas CA1508 thinks it is always false.

**Comment by mavasani** [comment]
Verified that this still repros on master
